ocaml/src/nsm_model.ml

```diff
@@ -32,6 +32,23 @@ module OsdInfo = struct
   type ip = string [@@deriving show, yojson]
   type port = int [@@deriving show, yojson]
 
+  type endpoint_uri = string [@@deriving show, yojson]
+
+  type host = string
+  type endpoint = Net_fd.transport * host * port
+
+  let parse_endpoint_uri s =
+    let uri = Uri.of_string s in
+    let transport = match Uri.scheme uri with
+      | None -> failwith "Transport should be 'tcp' or 'rdma'"
+      | Some "tcp" -> Net_fd.TCP
```

Comment:
Changed 'tcp' to 'tCp' in setup.ml and the test still seems to work, so it looks the library we use indeed does this conversion...
Fwiw see http://stackoverflow.com/questions/2148603/is-the-protocol-name-in-urls-case-sensitive